Home work1

deriglazoff opened the pull request.

deriglazoff committed:
SquareRootTest

deriglazoff committed:
Update dotnet.yml

all branch

deriglazoff committed:
nullable

deriglazoff committed:
Merge branch 'HomeWork1' of https://github.com/deriglazoff/space-battle into HomeWork1

a-poliakov commented on src/SpaceBattle.Api/SquareRoot.cs:
```diff
@@ -0,0 +1,33 @@
+            return [];
+        }
+
+        if (discriminant == 0)
```
стоит сравнивать с epsilon

a-poliakov commented on src/SpaceBattle.Api/SquareRoot.cs:
```diff
@@ -0,0 +1,33 @@
+            throw new ArgumentException($"{a} не равно 0", nameof(a));
+        }
+        var discriminant = Math.Pow(b, 2) - 4 * a * c;
+        if (discriminant < 0)
```
стоит сравнивать с epsilon

deriglazoff committed:
add double.NaN double.PositiveInfinity

deriglazoff committed:
epsilon

deriglazoff committed:
= epsilon

a-poliakov commented on src/SpaceBattle.Api/SquareRoot.cs:
```diff
@@ -0,0 +1,55 @@
+
+        var discriminant = Math.Pow(b, 2) - 4 * a * c;
+
+        if (Math.Abs(a) >= epsilon && Math.Abs(b) >= epsilon && Math.Abs(c) >= epsilon)
```
- если discriminant  > epsilon -> x1, x2
- если - epsilon <= discriminant  <= epsilon -> x1
- если discriminant  < -epsilon -> no roots

deriglazoff committed:
- если discriminant > epsilon -> x1, x2
- если -epsilon <= discriminant <= epsilon -> x1
- если discriminant < -epsilon -> no roots

deriglazoff committed:
movable and rotable

deriglazoff committed:
ICommand

deriglazoff committed:
ExceptionHandler

deriglazoff committed:
fix static

deriglazoff committed:
IMovable, IRotation

deriglazoff committed:
Merge pull request #3 from deriglazoff/hw3

ExceptionHandler

deriglazoff committed:
Merge pull request #2 from deriglazoff/hw2

movable and rotable
